Flake in `TestTxPool_ExpiredTxs_Timestamp`

## Description

We're currently experiencing a flake see https://github.com/celestiaorg/celestia-core/actions/runs/7742611329/job/21112148913?pr=1206
```
--- FAIL: TestTxPool_ExpiredTxs_Timestamp (0.01s)
    pool_test.go:581: Transaction 54784B65797B394244464131313144463739394636464239434641344439363445313846433330463434433434303534394537323146353131453138343334443337353342347D should still be in the mempool, but is not
    pool_test.go:581: Transaction 54784B65797B393631414535343545453445454633303133334239414141373743344144363236444644343537343338303936453043413744453433383843343032303433437D should still be in the mempool, but is not
    pool_test.go:581: Transaction 54784B65797B423632424531393838464339384644334145463835383639364441353535373033384634444644333934444338303632394643363139433639383334304136377D should still be in the mempool, but is not
    pool_test.go:581: Transaction 54784B65797B353438324541443442424139433638324533433642354134393737423646464233343331383633434642413232453639363432463530343333344636374346447D should still be in the mempool, but is not
    pool_test.go:581: Transaction 54784B65797B334244373936344230374237463542383046463830434339303341423337373538344238363838434531313033313845304241434642363438344333453742437D should still be in the mempool, but is not
    pool_test.go:581: Transaction 54784B65797B453834344233343638464638363932314532463736454231364635423338313538434233394438313136434245453133464531384246353332333141303841427D should still be in the mempool, but is not
FAIL
coverage: 86.6% of statements
FAIL	github.com/cometbft/cometbft/mempool/cat	18.396s
```

## Acceptance Criteria

- [ ] Investigate and resolve the flake

--- a/mempool/cat/pool_test.go
+++ b/mempool/cat/pool_test.go
@@ -553,16 +553,11 @@ func TestTxPool_ExpiredTxs_Timestamp(t *testing.T) {
 
 	// Wait a while, then add some more transactions that should not be expired
 	// when the first batch TTLs out.
-	//
-	// ms: 0   1   2   3   4   5   6
-	//     ^           ^       ^   ^
-	//     |           |       |   +-- Update (triggers pruning)
-	//     |           |       +------ first batch expires
-	//     |           +-------------- second batch added
-	//     +-------------------------- first batch added
-	//
-	// The exact intervals are not important except that the delta should be
-	// large relative to the cost of CheckTx (ms vs. ns is fine here).
+	// Because the TTL is 5ms which is very short, we need to have a more precise
+	// pruning interval to ensure that the transactions are expired
+	// so that the expired event is caught quickly enough
+	// that the second batch of transactions are not expired.
+
 	time.Sleep(2500 * time.Microsecond)
 	added2 := checkTxs(t, txmp, 10, 1)
 

--- a/mempool/cat/pool_test.go
+++ b/mempool/cat/pool_test.go
@@ -563,30 +563,35 @@ func TestTxPool_ExpiredTxs_Timestamp(t *testing.T) {
 	//
 	// The exact intervals are not important except that the delta should be
 	// large relative to the cost of CheckTx (ms vs. ns is fine here).
-	time.Sleep(3 * time.Millisecond)
+	time.Sleep(2500 * time.Microsecond)
 	added2 := checkTxs(t, txmp, 10, 1)
 
-	// Wait a while longer, so that the first batch will expire.
-	time.Sleep(3 * time.Millisecond)
+	// use require.Eventually to wait for the TTL to expire
+	require.Eventually(t, func() bool {
+		// Trigger an update so that pruning will occur.
+		txmp.Lock()
+		defer txmp.Unlock()
+		require.NoError(t, txmp.Update(txmp.height+1, nil, nil, nil, nil))
 
-	// Trigger an update so that pruning will occur.
-	txmp.Lock()
-	defer txmp.Unlock()
-	require.NoError(t, txmp.Update(txmp.height+1, nil, nil, nil, nil))
+		// All the transactions in the original set should have been purged.
+		for _, tx := range added1 {
+			// Check that it was added to the evictedTxCache
+			evicted := txmp.WasRecentlyEvicted(tx.tx.Key())
+			if !evicted {
+				return false
+			}
 
-	// All the transactions in the original set should have been purged.
-	for _, tx := range added1 {
-		// Check that it was added to the evictedTxCache
-		evicted := txmp.WasRecentlyEvicted(tx.tx.Key())
-		require.True(t, evicted)
-
-		if txmp.store.has(tx.tx.Key()) {
-			t.Errorf("Transaction %X should have been purged for TTL", tx.tx.Key())
+			if txmp.store.has(tx.tx.Key()) {
+				t.Errorf("Transaction %X should have been purged for TTL", tx.tx.Key())
+				return false
+			}
+			if txmp.rejectedTxCache.Has(tx.tx.Key()) {
+				t.Errorf("Transaction %X should have been removed from the cache", tx.tx.Key())
+				return false
+			}
 		}
-		if txmp.rejectedTxCache.Has(tx.tx.Key()) {
-			t.Errorf("Transaction %X should have been removed from the cache", tx.tx.Key())
-		}
-	}
+		return true
+	}, 10*time.Millisecond, 50*time.Microsecond)
 
 	// All the transactions added later should still be around.
 	for _, tx := range added2 {

